pkg/watch/health.go

```diff
@@ -85,44 +101,48 @@ func MonitorPodHealth(config *preparer.PreparerConfig, logger *logging.Logger, s
 // service it is monitoring. Every HEALTHCHECK_INTERVAL it
 // performs a health check and writes that information to
 // consul
-func (p *PodWatch) MonitorHealth(node string, store kp.Store, shutdownCh chan bool) {
+func (p *PodWatch) MonitorHealth(store kp.Store, statusChecker StatusCheck, shutdownCh chan bool) {
 	for {
 		select {
 		case <-time.After(HEALTHCHECK_INTERVAL):
-			p.checkHealth(node, p.manifest.StatusPort, store)
+			p.checkHealth(store, statusChecker)
 		case <-shutdownCh:
 			return
 		}
 	}
 }
 
-func (p *PodWatch) checkHealth(node string, port int, store kp.Store) {
-	resp, err := kp.HttpStatusCheck(node, port)
+func (p *PodWatch) checkHealth(store kp.Store, sc StatusCheck) {
+	resp, err := sc.Check()
 	health, err := resultFromCheck(resp, err)
 	if err != nil {
 		return
 	}
-	health.ID = p.manifest.Id
-	health.Node = node
-	health.Service = p.manifest.Id
+	health.ID = sc.ID
+	health.Node = sc.Node
+	health.Service = sc.ID
 
-	if p.updateNeeded(health, TTL) {
-		p.lastCheck, err = writeToConsul(health, store)
-		p.lastStatus = health.Status
+	if sc.updateNeeded(health, TTL) {
```

Comment:
I think there are two actions going on here:

* Get an app's status
* Write a status to Consul when needed

The procedures and data needed to do each action are completely separate; they don't gain anything by being bundled into the same structure. Can you break the latter task into its own unit?